branches.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"strings"
+
+	"github.com/rainforestapp/rainforest-cli/rainforest"
+	"github.com/urfave/cli"
+)
+
+type branchAPI interface {
+	CreateBranch(*rainforest.Branch) error
+}
+
+func newBranch(c cliContext, api branchAPI) error {
+	name := c.Args().First()
+	name = strings.TrimSpace(name)
```

Comment:
We might not need this line as it seems like the cli context already does it(which makes sense), but I wasn't able to find a confirmation of this. I didn't bother going trough the source for the sake of time. @magni- Any thoughts?

Reply:
We don't need this for (`$` to show EOL)
```
./rainforest branches new      some-branch-name    $
```
But if we want to disallow precending/trailing whitespace, we'll need it to trim cases such as
```
./rainforest branches new "  name with spaces before and after   "$
```
(but maybe that trimming should happen on the backend instead?)

Reply:
Good point on handling this in the backend. Will remove it here and add to the backend if we're not already doing it. 👍

Reply:
On second thought I'll leave this in to make the output nicer. `"branchName"` vs `" branchName "`.
Atm we don't do trimming on the backend. I'll add it there as well.